sentry_sdk/integrations/httpx.py

```diff
@@ -91,13 +97,17 @@ async def send(self, request, **kwargs):
             span.set_data("http.query", parsed_url.query)
             span.set_data("http.fragment", parsed_url.fragment)
 
-            for key, value in hub.iter_trace_propagation_headers():
-                logger.debug(
-                    "[Tracing] Adding `{key}` header {value} to outgoing request to {url}.".format(
-                        key=key, value=value, url=request.url
+            if should_propagate_trace(
+                str(request.url),
+                hub.client.options["trace_propagation_targets"] if hub.client else [],
+            ):
+                for key, value in hub.iter_trace_propagation_headers():
```

Comment:
`iter_trace_propagation_headers` is a hub method and already has some early return checks, so I would just add this logic there, and then we don't have to replicate it in every integration.

Reply:
sry ignore, i'm stupid